web/src/user_profile.ts

```diff
@@ -1171,7 +1171,21 @@ export function show_edit_user_info_modal(user_id: number, $container: JQuery):
         const user_id = Number($("#edit-user-form").attr("data-user-id"));
         function handle_confirm(): void {
             const url = "/json/users/" + encodeURIComponent(user_id);
-            dialog_widget.submit_api_request(channel.del, url, {});
+            const is_spammer = $("#is-user-spammer").is(":checked");
+
+            const NO_DELETE_ACTION = 0;
+            const DELETE_PUBLIC_STREAM_MESSAGES = 1;
+            const DELETE_ALL_MESSAGES = 2;
```

Comment:
I think this can be defined probably in `settings_config`. These are not related to settings, but we have other constants like these defined there. And then these constants can be directly used as value for the `option` elements instead of using strings like "delete-public", etc.